components/Poll/Poll.jsx

```diff
@@ -0,0 +1,342 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { View, FlatList, TouchableOpacity, Platform, Alert } from 'react-native';
+import { CheckBox } from 'react-native-elements';
+import { Text, Button, Textarea, Icon } from 'native-base';
+import { Font } from "expo";
+import { pollVote, pollOption } from '../../helpers/poll';
+import styles from "./PollStyle";
+
+export default class Poll extends React.Component {
+
+  constructor(props) {
+    super(props);
+
+    let cleanTitle = this.props.data && this.props.data.title ? this.props.data.title : '';
+    let cleanOpts = this.props.data && this.props.data.options ? this.props.data.options : [];
+    let cleanMultiSelect = this.props.data && this.props.data.multiSelect === false ? false : true;
+    let cleanOpen = this.props.data && this.props.data.open === true ? true : false;
+
+    this.state = {
+      title: cleanTitle,
+      multiSelect: cleanMultiSelect,
+      open: cleanOpen,
+      options: cleanOpts,
+      newOption: '',
+    }
+  }
+
+  async componentWillMount() {
+    await Font.loadAsync({
+      Roboto: require("native-base/Fonts/Roboto.ttf"),
+      Roboto_medium: require("native-base/Fonts/Roboto_medium.ttf")
+    });
+  }
+
+  newOptionUpdate = async (text) => {
+    if (text.length >= 200) {
+      return;
+    }
+    this.setState({ newOption: text });
+  }
+
+  titleUpdate = async (text) => {
+    if (text.length >= 1000) {
+      return;
+    }
+    this.state.title = text;
+    this.props.savePoll({
+      title: text,
+      multiSelect: this.state.multiSelect,
+      open: this.state.open,
+      options: this.state.options,
+    });
+  }
+
+  toggleMultiSelect = async () => {
+    if (this.state.multiSelect) {
+      for (var i = 0; i < this.state.options.length; i++) {
+        let allUsersVoted = this.state.options.reduce((acc, option) => acc.concat(option.usersVoted), []);
+        let hasDuplicates = ((new Set(allUsersVoted)).size !== allUsersVoted.length);
+        if (hasDuplicates) {
+          Alert.alert(
+            'Sorry',
```

Comment:
Note for myself and @CharlotteBaker316 that we should come up with a consistent/friendly way for error messages :)